### ISSUE TITLE
Random crashes when using database as session backend.

### ISSUE THREAD

**Issue by danslo** [danslo/hhvm#2]

**Comment by danslo** [comment]
Reproduce: 

``` php
<?php

require_once 'app/Mage.php';

Mage::init();

$session = Mage::getResourceSingleton('core/session');
$session->setSaveHandler();
```

Run it a couple of times in the <b>browser</b>.

**Comment by danslo** [comment]
PHP repro:

``` php
function open($savePath, $sessName) {}  
function close() {}
function read($sessId) {}
function write($sessId, $sessData) {}
function destroy($sessId) {}
function gc($sessMaxLifeTime) {}

session_set_save_handler('open', 'close', 'read', 'write', 'destroy', 'gc');
```

**Comment by danslo** [comment]
Official issue: https://github.com/facebook/hiphop-php/issues/990

**Comment by danslo** [comment]
Closed through 5c7ac5db43.
